### PR TITLE
Show create job errors in banner above job detail 

### LINKED ISSUE #122: show create job errors in banner above job detail
The backend currently returns a `status_message` field in the `DescribeJob` REST API endpoint, whic describes why job creation failed. This should be rendered on top of the job detail view in some kind of error banner component from MUI.

### PR DESCRIPTION
**Pull request by andrii-i**
Show create job errors in banner above job detail. 

Closes #122 

Preview:
no_title_under_buttonbar_cards_visible

Contrast rate measurements for alert bar via [monsindo](https://monsido.com/tools/contrast-checker), in line with [WCAG Contrast requirements (enchanced)](https://www.w3.org/WAI/WCAG21/quickref/#contrast-enhanced):
Screen Shot 2022-10-11 at 6 09 35 PM

### CODE REVIEW

File: src/mainviews/detail-view/job-detail.tsx [hunk 1]
```diff
@@ -268,6 +270,12 @@ export function JobDetail(props: IJobDetailProps): JSX.Element {
   return (
     <>
       {ButtonBar}
+      {props.model.statusMessage && (
+        <Alert severity="error">
+          <AlertTitle>Error</AlertTitle>
```
Comment (jweill-aws) [inline src/mainviews/detail-view/job-detail.tsx:275]: This should be translated using `trans.__()`.

File: src/model.ts [hunk 2]
```diff
@@ -139,10 +140,10 @@ export interface IJobDefinitionModel extends ICreateJobModel {
 }
 // Convert an IDescribeJobModel to an IJobDetailModel
 export function convertDescribeJobtoJobDetail(
-  dj: Scheduler.IDescribeJob
+  jd: Scheduler.IDescribeJob
```
Comment (jweill-aws) [inline src/model.ts:143]: `dj` stands for **D**escribe**J**ob; why change it to `jd`?

File: src/model.ts [hunk 3]
```diff
@@ -154,31 +155,32 @@ export function convertDescribeJobtoJobDetail(
...
+    updateTime: jd.update_time,
+    startTime: jd.start_time,
+    endTime: jd.end_time,
     scheduleInterval: 'weekday'
   };
 }
 
 export function convertDescribeDefinitiontoDefinition(
-  dj: Scheduler.IDescribeJobDefinition
+  jd: Scheduler.IDescribeJobDefinition
```
Comment (jweill-aws) [inline src/model.ts:180]: Again, the `jd` is ambiguous. Perhaps just renaming it to a word like `input` would be easier to understand.

File: src/model.ts (hunk 2 above)
Comment (andrii-i) [reply src/model.ts:143]: It makes sense. Somehow couldn't make sense of what `dj` stands for, thought it's a typo for `JobDefinition`

File: src/model.ts (hunk 3 above)
Comment (andrii-i) [reply src/model.ts:180]: Would `describeJob` be too verbose?

Comment (jweill-aws) [reply src/model.ts:180]: I don't think they're too verbose

File: src/mainviews/detail-view/job-detail.tsx (hunk 1 above)
Comment (jweill-aws) [inline src/mainviews/detail-view/job-detail.tsx:275]:
```suggestion
          <AlertTitle>{trans.__('Error')}</AlertTitle>
```

Comment (andrii-i) [reply src/mainviews/detail-view/job-detail.tsx]: Removed `AlertTitle` altogether

File: src/model.ts (hunk 3 above)
Comment (andrii-i) [reply src/model.ts]: inside `convertDescribeDefinitiontoDefinition ` renamed `jd`/`dj` to `describeDefinition`

File: src/model.ts (hunk 2 above)
Comment (andrii-i) [reply src/model.ts]: inside `convertDescribeJobtoJobDetail ` renamed `jd`/`dj` to `describeJob`

File: src/mainviews/detail-view/job-detail.tsx (hunk 1 above)
Comment (andrii-i) [reply src/mainviews/detail-view/job-detail.tsx]: For now, remove `AlertTitle ` altogether

File: src/model.ts [hunk 4]
```diff
@@ -138,75 +139,73 @@ export interface IJobDefinitionModel extends ICreateJobModel {
...
-    startTime: dj.start_time,
-    endTime: dj.end_time,
+    jobId: describeJob.job_id,
+    jobName: describeJob.name ?? '',
+    inputFile: describeJob.input_uri,
+    outputPath: describeJob.output_uri,
+    outputPrefix: describeJob.output_prefix,
+    environment: describeJob.runtime_environment_name,
+    parameters: jobParameters,
+    outputFormats: [],
```
Comment (jweill-aws) [inline src/model.ts:169]: Please rebase; `outputFormats` should now be set in `main`

File: src/model.ts [hunk 5]
```diff
@@ -138,75 +139,73 @@ export interface IJobDefinitionModel extends ICreateJobModel {
...
-    createTime: dj.create_time,
-    updateTime: dj.update_time,
-    schedule: dj.schedule,
-    timezone: dj.timezone,
+    definitionId: describeDefinition.job_definition_id,
+    outputPath: describeDefinition.output_filename_template ?? '',
+    outputPrefix: describeDefinition.output_prefix,
+    environment: describeDefinition.runtime_environment_name,
+    parameters: definitionParameters,
+    outputFormats: [],
```
Comment (jweill-aws) [inline src/model.ts:201]: Rebase; this is now set in `main`

Comment (andrii-i) [reply src/model.ts]: Done!

**Review by jweill-aws — APPROVED** [review]

**Pull request merged**